### PR TITLE
Docs: Update JavaScript build and setup with wp-scripts updates

### LINKED ISSUE #13712: Update JS Build instructions to use wp-scripts
Update JS Build instructions to use wp-scripts package when published after 5.1 release.

The updated wp-scripts will include a `build` and `start` scripts that allow simplifying the JS build instructions.

Per #13629

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Closes #13712.

Updates the build and setup section of the JavaScript tutorial with the new simplified commands included in wp-scripts. Removes webpack and babel configs and replaces recommending to use the default configs using `wp-scripts build|start` 

## How has this been tested?

Follow the instructions, in the tutorial which includes a simple block you can create to test.
Until the wp-scripts gets published you will need to add a local reference to `@wordpress/scripts` package. 

For me Gutenberg is one directory up from my sample plugin so my package.json entry looks like:
```
    "@wordpress/scripts": "file:../gutenberg/packages/scripts",
```

View [documentation live on branch here](https://github.com/WordPress/gutenberg/blob/update/js-build-docs/docs/designers-developers/developers/tutorials/javascript/js-build-setup.md)

Depends on #14168

### CODE REVIEW

**Comment by youknowriad** [comment]
Nice ❤️

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]:
```suggestion
The `@wordpress/scripts` package abstracts these libraries away to standardize and simplify development, so you won't need to handle the details for configuring those libraries.
```

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]:
```suggestion
After installing, a `node_modules` directory is created with the modules and their dependencies.
```

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]:
```suggestion
The `@wordpress/scripts` package handles the default configuration for Webpack and Babel. You need to add the following packages:
```

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]:
```suggestion
The scripts package expects the source file to compile to be found at `src/index.js`, and will save the compiled output to `build/index.js`.

With that in mind, let's set up a basic block. Create a file at `src/index.js` with the following content:
```

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]:
```suggestion
The **build** command in `@wordpress/scripts` runs in a "production" mode. This shrinks the code down so it downloads faster, but makes it difficult to read in the process. You can use the **start** command which runs a development mode that does not shrink the code, and additionally continues a running process to watch the source file for more changes and rebuild as you develop.
```

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]:
```suggestion
The start command can be added to the same scripts section of `package.json`:
```

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (chrisvanpatten) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]:
```suggestion
Yes, the initial setup is a bit more involved, but the additional features and benefits are usually worth the trade off in setup time.
```

**Review by chrisvanpatten — DISMISSED** [review]
Looks great! A few small suggestions…

**Comment by mkaz** [comment]
@nosolosw @gziolo In my testing, it was still required to install the following packages, they were not picked up as dependencies of @wordpress/scripts
 
```webpack webpack-cli @wordpress/babel-preset-default babel-plugin-transform-react-jsx```

The `babel-plugin-transform-react-jsx` looks to be the only one not specified in the scripts package.json. I'm not sure why the others didn't work for me, my hunch is because of the local package references.

Following the instructions in the docs and installing and it all works great, so much simpler!

**Comment by nosolosw** [comment]
@mkaz for ESNext we need to land https://github.com/WordPress/gutenberg/pull/14168 first. I'm waiting for Travis so I can merge :)

This example https://github.com/nosolosw/understanding-gutenberg/pull/7 shows how it works (I'm using JSX syntax and arrow functions in the `src/index.js` and the `build/index.js` is transpiled properly).

**Review by oandregal — APPROVED** [review]
This is great to see! I pushed some commits, feel free to remove them if you don't like them. I thought it'd be faster to commit than to write the suggestion :)

I'm approving so it's ready to go, although I understand that we want to merge this after the packages' release.

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (gziolo) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]: Should we also link npm package?
```suggestion
The [@wordpress/scripts](https://www.npmjs.com/package/@wordpress/scripts) package abstracts these libraries away to standardize and simplify development, so you won't need to handle the details for configuring those libraries.
```

File: docs/designers-developers/developers/tutorials/javascript/js-build-setup.md
Comment (gziolo) [inline docs/designers-developers/developers/tutorials/javascript/js-build-setup.md]: Next step, hot module reloading support? :)

**Review by gziolo — APPROVED** [review]
Awesome work. Thanks for putting this together and updating with the latest changes introduced. It should be very helpful moving forward. We probably should also reference this document from `@wordpress/scripts` readme file as well.

**Pull request merged**
